staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -73,17 +74,28 @@ var (
 
 // shouldHide be used to check if a specific metric with deprecated version should be hidden
 // according to metrics deprecation lifecycle.
-func shouldHide(currentVersion *semver.Version, deprecatedVersion *semver.Version) bool {
+// This follows rule #11b as stated here: https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-metric
+func shouldHide(stabilityLevel StabilityLevel, currentVersion, deprecatedVersion *semver.Version) bool {
 	guardVersion, err := semver.Make(fmt.Sprintf("%d.%d.0", currentVersion.Major, currentVersion.Minor))
```

Comment:
I think we want to make the boundary alpha.1 which happens ~3 weeks into the release cycle.

Otherwise unfortunately we wind up with the metric being disabled on master when we cut the release branch (master becomes n+1 alpha, so right now 1.35.0-alpha...), and some test breaking in a way we don't want to backport to the ongoing release while we're still fast-forwarding (eg we remove the test because the metric is not valid anymore in 1.35+).

This is how the API removals work, they actually drop in alpha.1+ (links in the bug discussions)

Reply:
https://github.com/kubernetes/kubernetes/blob/ee065dfc80b5762d53c5d9d896a8cc4b1d9c6bd2/staging/src/k8s.io/apiserver/pkg/server/deleted_kinds.go#L161

Reply:
Understood, it seems I pushed the latest commit prematurely. I'll make some adjustments to set the boundary to `alpha.1` and add some tests.